tests\drivers\build_all\ethernet build fail on s32z270dc2 boards

**Describe the bug**

tests\drivers\build_all\ethernet build fail on s32z270dc2 boards

**To Reproduce**

1. `west build -p -b s32z270dc2_rtu0_r52 .\tests\drivers\build_all\ethernet`
2. See the error log

**Expected behavior**
No build issues

**Impact**
Main development blocked when this test is triggered in CI

**Logs and console output**
```
[126/224] Building C object zephyr/drivers/mdio/CMakeFiles/drivers__mdio.dir/mdio_nxp_s32_netc.c.obj
FAILED: zephyr/drivers/mdio/CMakeFiles/drivers__mdio.dir/mdio_nxp_s32_netc.c.obj 
D:\tools\zephyr-sdk-0.16.3_windows-x86_64\zephyr-sdk-0.16.3\arm-zephyr-eabi\bin\arm-zephyr-eabi-gcc.exe -DCPU_S32Z27 -DFSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL=1 -DKERNEL -DPICOLIBC_DOUBLE_PRINTF_SCANF -D_FORTIFY_SOURCE=1 -D_POSIX_C_SOURCE=200809 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -ID:/upstream/zephyr/include -ID:/upstream/zephyr/build/zephyr/include/generated -ID:/upstream/zephyr/soc/arm/nxp_s32/s32ze -ID:/upstream/zephyr/soc/arm/nxp_s32/common/. -ID:/upstream/zephyr/subsys/testsuite/include -ID:/upstream/zephyr/subsys/net/l2 -ID:/upstream/zephyr/subsys/net/lib/dns/. -ID:/upstream/modules/hal/cmsis/CMSIS/Core_R/Include -ID:/upstream/zephyr/modules/cmsis/. -ID:/upstream/modules/hal/nxp/s32/drivers/s32ze/BaseNXP/header -ID:/upstream/modules/hal/nxp/s32/drivers/s32ze/BaseNXP/include -ID:/upstream/modules/hal/nxp/s32/drivers/s32ze/Mcu/include -ID:/upstream/modules/hal/nxp/s32/drivers/s32ze/Port/include -ID:/upstream/modules/hal/nxp/s32/drivers/s32ze/Rte/include -ID:/upstream/modules/hal/nxp/s32/drivers/s32ze/Uart/include -ID:/upstream/modules/hal/nxp/s32/soc/s32z27/include -ID:/upstream/modules/hal/nxp/s32/mcux/devices/S32Z27 -ID:/upstream/modules/hal/nxp/s32/mcux/devices/S32Z27/drivers -ID:/upstream/modules/hal/nxp/mcux/mcux-sdk/drivers/common -ID:/upstream/zephyr/modules/hal_nxp/. -Wshadow -fno-strict-aliasing -Os -imacros D:/upstream/zephyr/build/zephyr/include/generated/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-r52 -mabi=aapcs -mfpu=neon-fp-armv8 -mfloat-abi=hard -mfp16-format=ieee -mtp=soft --sysroot=D:/tools/zephyr-sdk-0.16.3_windows-x86_64/zephyr-sdk-0.16.3/arm-zephyr-eabi/arm-zephyr-eabi -imacros D:/upstream/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=D:/upstream/zephyr/tests/drivers/build_all/ethernet=CMAKE_SOURCE_DIR -fmacro-prefix-map=D:/upstream/zephyr=ZEPHYR_BASE -fmacro-prefix-map=D:/upstream=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/mdio/CMakeFiles/drivers__mdio.dir/mdio_nxp_s32_netc.c.obj -MF zephyr\drivers\mdio\CMakeFiles\drivers__mdio.dir\mdio_nxp_s32_netc.c.obj.d -o zephyr/drivers/mdio/CMakeFiles/drivers__mdio.dir/mdio_nxp_s32_netc.c.obj -c 
D:/upstream/zephyr/drivers/mdio/mdio_nxp_s32_netc.c
D:/upstream/zephyr/drivers/mdio/mdio_nxp_s32_netc.c:13:10: fatal error: Netc_EthSwt_Ip.h: No such file or directory
   13 | #include <Netc_EthSwt_Ip.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[133/224] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/ipv6_nbr.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'D:\upstream\zephyr\build'
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used 8bec2d7b37e14b5605f786f3ea6c0d3749fc96ab

**Additional context**
The issue is seen after this commit https://github.com/zephyrproject-rtos/zephyr/commit/00d03fe942c08c2444636d24266333cfcaea5fcb.

Root cause:
 - After the commit got merged, the `CONFIG_MDIO` is selected when `CONFIG_ETH_ADIN2111` is enabled.
 - `CONFIG_MDIO=y` --> `CONFIG_MDIO_NXP_S32_NECT=y` (because mdio node is enabled in boards devicetree), but in order to get the mdio nxp s32 netc driver build, `CONFIG_ETH_NXP_S32_NETC` must be set, but currently is not because of this dependency https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/ethernet/Kconfig.nxp_s32_netc#L8

--- a/drivers/ethernet/Kconfig.nxp_s32_netc
+++ b/drivers/ethernet/Kconfig.nxp_s32_netc
@@ -1,11 +1,10 @@
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 # SPDX-License-Identifier: Apache-2.0
 
 menuconfig ETH_NXP_S32_NETC
 	bool "NXP S32 Ethernet Switch and Controller (NETC) driver"
 	default y
 	depends on (DT_HAS_NXP_S32_NETC_PSI_ENABLED || DT_HAS_NXP_S32_NETC_VSI_ENABLED)
-	depends on !NET_TEST
 	select MBOX
 	select MDIO if DT_HAS_NXP_S32_NETC_PSI_ENABLED
 	select NOCACHE_MEMORY if ARCH_HAS_NOCACHE_MEMORY_SUPPORT

--- a/drivers/mdio/Kconfig.nxp_s32
+++ b/drivers/mdio/Kconfig.nxp_s32
@@ -1,10 +1,11 @@
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 # SPDX-License-Identifier: Apache-2.0
 
 config MDIO_NXP_S32_NETC
 	bool "NXP S32 NETC External MDIO driver"
 	default y
 	depends on DT_HAS_NXP_S32_NETC_EMDIO_ENABLED
+	depends on ETH_NXP_S32_NETC && DT_HAS_NXP_S32_NETC_PSI_ENABLED
 	select NOCACHE_MEMORY if ARCH_HAS_NOCACHE_MEMORY_SUPPORT
 	help
 	  Enable NETC External MDIO Controller driver for NXP S32 SoCs.

